index.js

```diff
@@ -23,32 +24,50 @@ class ForgeExternalsPlugin {
     switch (hookName) {
       case "resolveForgeConfig":
         return this.resolveForgeConfig;
+      case "packageAfterCopy":
+        return this.packageAfterCopy;
     }
   }
 
   getHooks() {
     return {
-      "resolveForgeConfig": this.resolveForgeConfig
+      "resolveForgeConfig": this.resolveForgeConfig,
+      "packageAfterCopy": this.packageAfterCopy,
     };
   }
 
   resolveForgeConfig = async (forgeConfig) => {
     const foundModules = new Set(this._externals);
+    this.modulesToCopy = {};
 
     if (this._includeDeps) {
       for (const external of this._externals) {
         const moduleRoot = dirname(
           require.resolve(`${external}/package.json`, { paths: [this._dir] })
         );
 
+        const rootNodeModules = dirname(moduleRoot);
+        // Add the external module to copy list
+        this.modulesToCopy[external] = moduleRoot;
+
         const walker = new Walker(moduleRoot);
         // These are private so it's quite nasty!
         walker.modules = [];
         await walker.walkDependenciesForModule(moduleRoot, DepType.PROD);
         walker.modules
           .filter((dep) => dep.nativeModuleType === DepType.PROD)
-          .map((dep) => dep.name)
-          .forEach((name) => foundModules.add(name));
+          .forEach((dep) => {
+            // How many occurences of `node_modules` in the path
+            const levels = dep.path.split(`${sep}node_modules${sep}`).length - 1;

```

Comment:
while a little naive, finding nested modules by splitting on `node_modules` err's on the side of copying too much (less bad than not enough)